interface/src/scripting/AssetMappingsScriptingInterface.cpp

```diff
@@ -46,28 +48,63 @@ void AssetMappingsScriptingInterface::getMapping(QString path, QJSValue callback
     auto request = assetClient->createGetMappingRequest(path);
 
     connect(request, &GetMappingRequest::finished, this, [this, callback](GetMappingRequest* request) mutable {
-        QJSValueList args { uint8_t(request->getError()), request->getHash() };
-
-        callback.call(args);
+        if (callback.isCallable()) {
+            QJSValueList args { uint8_t(request->getError()) };
+            callback.call(args);
+        }
 
         request->deleteLater();
-
     });
 
     request->start();
 }
 
+void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping, QJSValue callback) {
+    auto offscreenUi = DependencyManager::get<OffscreenUi>();
+    auto result = offscreenUi->inputDialog(OffscreenUi::ICON_NONE, "Enter asset path:", "", mapping);
+    if (!result.isValid()) {
+        return;
+    }
+    mapping = result.toString();
+
+    // Check for override
+    if (isKnownMapping(mapping)) {
+        auto message = "The following file already exists:\n" + path + "\nDo you want to override it?";
+        auto button = offscreenUi->messageBox(OffscreenUi::ICON_QUESTION, "", message,
+                                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
+        if (button == QMessageBox::No) {
+            return;
+        }
+    }
+
+    auto upload = DependencyManager::get<AssetClient>()->createUpload(path);
+    QObject::connect(upload, &AssetUpload::finished, this, [=](AssetUpload* upload, const QString& hash) mutable {
+        if (upload->getError() != AssetUpload::NoError) {
+            if (callback.isCallable()) {
+                QJSValueList args { uint8_t(upload->getError()) };
+                callback.call(args);
+            }
+        } else {
+            setMapping(mapping, hash, callback);
```

Comment:
There's probably not a good short term solution to this since the client doesn't know the hashes of the assets that are already present in the asset server